luminoso_api/v4_client.py

```diff
@@ -374,60 +99,32 @@ def get_raw(self, path, **params):
         url = ensure_trailing_slash(self.url + path.lstrip('/'))
         return self._request('get', url, params=params).text
 
-    def save_to_file(self, path, filename, **params):
+    @staticmethod
+    def get_root_url(url, warn=True):
         """
-        Saves binary content to a file with name filename. filename should
-        include the appropriate file extension, such as .xlsx or .txt, e.g.,
-        filename = 'sample.xlsx'.
+        Get the "root URL" for a URL, as described in the LuminosoClient
+        documentation.
         """
-        url = ensure_trailing_slash(self.url + path.lstrip('/'))
-        content = self._request('get', url, params=params).content
-        with open(filename, 'wb') as f:
-            f.write(content)
+        parsed_url = urlparse(url)
 
+        # Make sure it's a complete URL, not a relative one
+        if not parsed_url.scheme:
+            raise ValueError('Please supply a full URL, beginning with http://'
+                             ' or https:// .')
 
-def get_token_filename():
-    """
-    Return the default filename for storing API tokens.
-    """
-    return os.path.join(os.path.expanduser('~'),
-                        '.luminoso',
-                        'tokens.json')
+        # Issue a warning if the path didn't already start with /api/v4
+        root_url = '%s://%s/api/v4' % (parsed_url.scheme, parsed_url.netloc)
+        if warn and not parsed_url.path.startswith('/api/v4'):
+            logger.warning('Using %s as the root url' % root_url)
+        return root_url
 
-def get_root_url(url, warn=True):
-    """
-    Get the "root URL" for a URL, as described in the LuminosoClient
-    documentation.
-    """
-    parsed_url = urlparse(url)
 
-    # Make sure it's a complete URL, not a relative one
-    if not parsed_url.scheme:
-        raise ValueError('Please supply a full URL, beginning with http:// '
-                         'or https:// .')
+get_root_url = LuminosoClient.get_root_url
 
-    # Issue a warning if the path didn't already start with /api/v4
-    root_url = '%s://%s/api/v4' % (parsed_url.scheme, parsed_url.netloc)
-    if warn and not parsed_url.path.startswith('/api/v4'):
-        logger.warning('Using %s as the root url' % root_url)
-    return root_url
 
 def ensure_trailing_slash(url):
```

Comment:
It's really tempting to inline it, but it's also possible that someone is importing this, so (for the same reason that, when I moved `get_root_url` onto the client so that it could be used by the superclass's `__init__` as `self.get_root_url`, I added `get_root_url = LuminosoClient.get_root_url` to keep from breaking imports) it's probably safer to leave it defined external to the class.

Reply:
Yeah, I figured as much, it just seems like an unlikely thing for someone to be using in general.  Fine with me to leave it where it is, though.  Hopefully the whole v4 client goes away in the not-too-distant future anyway.